### PR TITLE
Add Thirdweb Create (Port over create-tw-app to thirdweb-cli)

### PR DESCRIPTION
**Pull request by ndeto**
Moving `create-tw-app` to thirdweb CLI

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,20 +30,28 @@
...
     "ethers": "^5.6.9",
     "form-data": "^4.0.0",
+    "got": "11.8.5",
     "inquirer": "^8.2.3",
     "ipfs-unixfs": "^6.0.7",
     "ipld-dag-pb": "^0.22.3",
     "isomorphic-fetch": "^3.0.0",
     "multihashes": "^4.0.3",
     "open": "^8.4.0",
+    "prompts": "^2.4.2",
```
Comment (joaquim-verges) [inline package.json:47]: we already have 2 prompt libraries xD

i want to only keep `enquirer` which is the one i liked the most. 

I guess we can do it as a follow up

File: package.json [hunk 2]
```diff
@@ -24,20 +30,28 @@
...
+    "rimraf": "^3.0.2",
     "solc": "0.8.13",
+    "tar": "4.4.18",
     "tslog": "^3.3.3",
+    "update-check": "^1.5.4",
     "update-notifier": "^5.1.0",
+    "validate-npm-package-name": "^4.0.0",
```
Comment (joaquim-verges) [inline package.json:54]: what's this for? @nachoiacovino

File: src/cli/index.ts [hunk 3]
```diff
@@ -77,10 +79,85 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
...
+      `
+
+  Explicitly tell the CLI to bootstrap the app using npm
+`,
+    )
+    .option(
+      "--use-pnpm",
+      `
+
+  Explicitly tell the CLI to bootstrap the app using pnpm
```
Comment (joaquim-verges) [inline src/cli/index.ts]: any reason why we need those line breaks here? does it look nicer in terminal or something?

File: src/cli/index.ts [hunk 4]
```diff
@@ -77,10 +79,85 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
       open(url.toString());
     });
 
+  program
+    .command("create")
+    .description(
+      "Compile & deploy contracts through your thirdweb dashboard, without dealing with private keys.",
```
Comment (joaquim-verges) [inline src/cli/index.ts]: wrong description

File: src/common/twCreate.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env node
```
Comment (joaquim-verges) [inline src/common/twCreate.ts]: would move this file to src/create

File: src/create/helpers/create-app.ts [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  console.log("We suggest that you begin by typing:");
+  console.log();
+  console.log(chalk.cyan("  cd"), cdpath);
+  console.log(
+    `  ${chalk.cyan(
+      `${packageManager} ${useYarn || startOrDev === "start" ? "" : "run "
+      }${startOrDev}`,
+    )}`,
+  );
+  console.log();
```
Comment (joaquim-verges) [inline src/create/helpers/create-app.ts:198]: eventually we'll want to switch to `logger` so we can control it better, but can be done later

File: src/create/helpers/examples.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      name,
+    )}/contents/package.json`,
+  );
+}
+
+export function downloadAndExtractRepo(
+  root: string,
+  { name, filePath }: RepoInfo,
+): Promise<void> {
+  return pipeline(
```
Comment (joaquim-verges) [inline src/create/helpers/examples.ts:31]: this is nice

File: src/create/helpers/examples.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    got.stream(
```
Comment (joaquim-verges) [inline src/create/helpers/examples.ts:32]: is this how vercel does it? surprised it doesn't just git clone

File: src/create/helpers/get-start-or-dev.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import got from "got/dist/source";
+
+export const getStartOrDev = async (name: string) => {
+  const { data } = await got
```
Comment (joaquim-verges) [inline src/create/helpers/get-start-or-dev.ts:4]: wonder if we should just use `fetch` here to not have 2 different http libraries

File: src/create/helpers/is-online.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+  try {
+    const httpsProxy = execSync("npm config get https-proxy").toString().trim();
+    return httpsProxy !== "null" ? httpsProxy : undefined;
+  } catch (e) {
+    return;
+  }
+}
+
+export function getOnline(): Promise<boolean> {
```
Comment (joaquim-verges) [inline src/create/helpers/is-online.ts:18]: handy

**Review by joaquim-verges — COMMENTED** [review]
mainly minor things - no need to address them now but stuff to think about and some questions for @nachoiacovino 

will test it out shortly

File: package.json (hunk 2 above)
Comment (nachoiacovino) [reply package.json:54]: this is to validate if it's a valid npm package name :D

**Review by joaquim-verges — APPROVED** [review]
ship it - let's make sure we update portal + guides (if any)

File: src/create/command.ts [hunk 11]
```diff
@@ -142,7 +137,7 @@ export async function twCreate(options: any) {
 
   if (options.example === true) {
     console.error(
-      "Please provide an example name, otherwise remove the example option.",
+      "Please provide an example name, otherwise remove the example option. Checkout some examples you can use here: https://github.com/thirdweb-example/",
```
Comment (joaquim-verges) [inline src/create/command.ts:140]: very nice, can we add the link to `option` description as well?

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**